### DISCUSSION TITLE
Version 2.1

### DISCUSSION THREAD

**Discussion by vkbo** [vkbo/novelWriter#1534] [General]
### Release Notes

The primary focus of this release has been a complete redesign of the Build Tool, that is, the tool that assembles your project into a manuscript document. The new tool, called the "Manuscript Build Tool" allows you to define multiple build definitions for your project. The build definitions are edited in a new Manuscript Build Settings dialog, with a lot more options than the old tool.

The reason for this redesign is a long list of feature requests that could not easily be accommodated in the old, much simpler tool. Far from all the features have been added yet, but now that the new tool is in place, they will be gradually added in the coming releases.

The key feature added in this release is the extended control you now have for selecting exactly what part of your project is included in a given build definition. You have the same filters for selecting documents and notes, and turning on or off root folders as before, but you can now easily override on a per-document basis what is included or excluded in addition to the filter.

A second major improvement is a better tool to format your manuscript headings. You no longer have to look up formatting codes and add them manually. Instead, there is now a heading format editor in the Build Settings dialog for creating the header format, with syntax highlighting included.

#### Other Changes

Among other features is a new option to duplicate documents and folders in the project tree. The duplicate feature is available from the right-click menu. A proper light colour theme has also been added. In most cases it will be the same as the default theme, depending on your platform.

There are other, minor improvements as well, and a lot of code improvements under the hood. For a full list of changes, see the detailed changelogs.

_These Release Notes also include the changes from the 2.1 Beta 1 and 2.1 RC 1 releases._

### Detailed Changelog

**Usability**

* A widget has been added to the Build Manuscript tool main window to show some select build settings for the selected build definition. This should make it a little easier to find the wanted build definition if there are many available. PR #1516.
* All columns on the Writing Stats tool now uses the same fixed width font. Issue #1442, PR #1518.

**Documentation**

* The documentation has received significant updates for the 2.1 release. PR #1531.

**Packaging and Installation**

* Python 3.7 support has officially been dropped. Python 3.7 has reached end of life, and dropping it relaxes some restrictions on development. PR #1515.
* MacOS and Windows is now tested against Python 3.11, and 3.12 has been added to Linux. PR #1515.

<hr /><em>This discussion was created from the release <a href='https://github.com/vkbo/novelWriter/releases/tag/v2.1'>Version 2.1</a>.</em>

**Comment by LegolasMirkwood** [comment]
2.1   I can use it well enough, the essentials anyways. Calibre is still butchering the double L's in the .pdfs made in Novelwriter Linux, just one L and a space when converting the .pdf to an .epub.  I DID find a workaround for that using Linux Libre Office Writer!  Build the document, copy all, paste into Libre Writer, export as .epub.  I haven't got the Cover.jpg to work from that, yet, BUT!  Converting the .epub in Calibre to .epub lets me fill in all the extras - cover.jpg, author, etc and builds it to the Library folder of Calibre.  Bit of Banging things around and it works, now to the important stuff.

Can't update the right 1/3 of the screen to the right tag reference frame of the screen without r/clicking a tag and "follow tag" it.  That's the only thing I've found in the Linux version to make that crucial right 1/3 of the screen useful and put the primary writing screen where it should be.  Which is never in the right place by default and this is the only way to get it there.

All the same,  Happy Halloween:  Every writer's first story is about sex.  If you've hit puberty and you can put a sentence together on paper?  Demons and things incoming for Halloween avert your eyes, I beg of you :)

[The Contract - Joyeuse Tyrfing.pdf](https://github.com/vkbo/novelWriter/files/13173612/The.Contract.-.Joyeuse.Tyrfing.pdf)

**Reply by vkbo** [reply]
I'm glad using Open Document + Libre Office works. It is the most feature complete export option from novelWriter. The PDF option is purely a print to file feature offered by the Qt framework novelWriter is built on, and most of their export features are fairly poor quality. They too have an Open Document export, but I don't use it as it doesn't even set headings correctly.

Not sure what you mean about opening tags. You can hit `Ctrl+Enter` when the cursor is on a tag, or hold `Ctrl` while clicking with the mouse on it. Pressing `Ctrl+R` while the editor has focus also opens the same document on the right.

**Reply by LegolasMirkwood** [reply]
Straight up, my compliments to the chef.  It isn't so much the little complications on the way as it is "The sheer Fuckin Focus, of John Wick". 
I admire what she's done.  I'd join that 3-some so fast my dick would be confused :D

ctrl+R renders a copy the story and not the People.  The people ARE the story.  A solid background for the Characters, bit of plot, bit of place.  That's what you need:

# Main

@tag:Main

Neal Smith: 40 yr old, 5’ 8” Blocky build - Heavy gym lifter, brawler, biker buds, Gay top, circumcised cock 7” Thick, Bushy, bear type in general, brown hair.  Casual drugs.  Relationship with Teak Acknowledged for the sole reason:  Nobody could beat him down to the ground.  Hells Raiders teen bike gang, now older motorcycle gang.  Red horn tattoos on shoulder blades no other tattoos.

Teak:  30yr old, Neal’s Vietnamese life partner, 5’ 5” slender build, sumpremely athletic, scrappy 0 to fast, extremely fast, 5” uncircumcised slender cock, black hair, olive skin, cocaine and speed - drugs of preference.  Red horn tattoos on shoulder blades.  R. sleeve tattoo, expert throwing knives.  Grew up in same neighborhood as Neal.  Many fights.  Broke Neal’s teak wood chessboard over a head in a brawl.  His only name is Teak, nobody remembers his born name.  

Sakiko Kikuchi: 20yr old, waiflike and innocent.  Neal’s wife, maiden name.  Last direct descendant bloodline female Kikuchi clan.  Sent to America to be left out of clan rivalry by her father, head of the KiKuchi clan.  Dainty red lotus tattoo on inner left thigh.  Rumored vampiric bloodline.  Teak calls her Sake, Neal calls her Sakiko in full and slowly, always.

Writing stories are like DMing w/out the rastafarians raising hell at a natural 20.  "You made the jump and now your balls are on fire, specifically the hair on you balls.  (Roll) which had caught the hair on your ass on fire."  Simply enjoying the time we're given :)

I could do another Character layout with Baribies, but, you get the idea.  Still HAVE to create tags and follow them.

**Reply by vkbo** [reply]
> ctrl+R renders a copy the story and not the People. The people ARE the story. A solid background for the Characters, bit of plot, bit of place. That's what you need:

My point was that there are several ways to open the notes, which is how I understood the first post. I'm not entirely sure what the issue is still. Yes, you have to create the tags for each character you want to reference. The reason for this is that you can then decide yourself where you want them to point to. You can define all your characters in a single file, and have a tag for each of them. As long as you add a header for each.

I have rewritten the documentation for this in 2.1, so hopefully it's clearer now.

**Comment by LegolasMirkwood** [comment]
Demo: This is how "I" use novelwriter 2.1.  For this brief moment in time.  I just feel you should know what it can be.  OK, lowkey - for the love of God, put, in a default layout similar to this.  A Tag.  A tag follow.  Template.  I figure things out most people don't.

I watched an hour long video to figure out this one simple thing that made everything fit right on my screen.  Hook a brotha up?

https://imgur.com/9GFYLhk
